### PR TITLE
Command line: adding variants for Require, aligning on the vernac syntax

### PR DESCRIPTION
**Pull request by herbelin**
We deprecate `-require` command line option (= `Require Import`) to avoid the confusion with `Require` (no import).

We propose a regular equivalent to all vernac variants in expanded and short version:
- `-require-import`, `-require-export`, `-require-import-from`, `-require-export-from`
- `-ri`, `-re`, `-rifrom`, `-refrom`

We also add `-rfrom`, but wait for the end of deprecation of `-require` to
replace `-load-vernac-object` (which indeed means `Require` w/o import) by `-require` and to introduce a shorthand `-r` for the new `-require`.

**Kind:** feature / bug-fix

This is the main user-visible feature part of #10147. See also #9585.

One may wonder whether these options are very useful. Personally, I found them useful (quicker for debugging to have a command `coqtop -require foo` in the shell history than typing `Require Import foo` in each  launch of a session), but that may be an isolated need.

In any case, the syntactic mismatch between `-load-vernac-object` and its semantcs `Require` and between `-require` and its semantics `Require Import` is certainly not good to keep. So, something has to be done anyway.

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by Zimmi48** [comment]
The fact that OCaml has `-open` seems to reinforce the idea that these commands are useful.

File: toplevel/coqargs.ml
Comment (ejgallego) [inline toplevel/coqargs.ml]: Maybe that's a good moment to turn `(Some false)` into `~scope:Import` ?

**Review by ejgallego — APPROVED** [review]
LGTM, thanks!

One question I have w.r.t. to short vs long options is whether we would like to use the `--require-import` form, which seems in general more standard if a short form is also provided.

I am not sure indeed about the use case for `Require` for supporting it is quite cheap so why not?

**Comment by ejgallego** [comment]
I'll merge when the docs are in place, IIUC.

File: toplevel/coqargs.ml
Comment (herbelin) [reply toplevel/coqargs.ml]: We could, but it is quite a lot of changes, up to `library.ml`. So, maybe later.

Comment (ejgallego) [reply toplevel/coqargs.ml]: Ok then.

**Comment by herbelin** [comment]
> we would like to use the `--require-import` form, which seems in general more standard if a short form is also provided.

Why not a double-dash indeed? However that's not much the usage for Coq. To help people used to GNU style, we could have both single dash `-require-import` and double-dash `--require-import`...

> I'll merge when the docs are in place, IIUC.

I added the options in the reference manual, but I'm unsure about the style. For instance, after the description of the command option `:-require-import *qualid*:`, should I use ``:n:`@qualid` `` or `*qualid*` for the explanation (maybe @Zimmi48 or @cpitclaudel can check).

**Comment by ejgallego** [comment]
> Why not a double-dash indeed? However that's not much the usage for Coq. To help people used to GNU style, we could have both single dash `-require-import` and double-dash `--require-import`...

I don't personally care a lot, however most libraries for cmline parsing in OCaml assume long options start with `--` so I dunno.

> I added the options in the reference manual,

Thanks @herbelin , let's wait to hear from @coq/doc-maintainers , rest seems good so once doc has been reviewed I'll merge.

**Comment by ejgallego** [comment]
Hi @coq/doc-maintainers could you have a look to this so I can merge?

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (cpitclaudel) [inline doc/sphinx/practical-tools/coq-commands.rst]: You can use ``:token:`dirpath` `` here, I think.
Also, there's a missing space before `Note that only…`

**Review by cpitclaudel — APPROVED** [review]

**Comment by ejgallego** [comment]
Thanks @cpitclaudel , @herbelin sorry to bother you, would you mind fixing the typo?

I'll merge afterwards.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (herbelin) [reply doc/sphinx/practical-tools/coq-commands.rst]: > You can use ``:token:`dirpath` ``here, I think.

In the same file, there are plenty of ``:n:@`foo` `` and no ``:token:`foo` ``. It seems better to me to keep the file consistent in style.

>  Also, there's a missing space before `Note that only`…

Thanks, I should have noticed it in passing. Fixed.

**Pull request merged**
